pygments/lexers/qlik.py

```diff
@@ -0,0 +1,139 @@
+            (r"\*/", Comment.Multiline, "#pop"),
+            (r"[^\*/]+", Comment.Multiline),
+        ],
+        # Handle numbers
+        "numerics": [
+            (r"\b[0-9][0-9]*\.[0-9]+([eE][0-9]+)?[fd]?\b", Number.Float),
+            (r"\b[0-9]+\b", Number.Integer),
+        ],
+        # Handle variable names in things
+        "interp": [
+            (r"(\$\()(\w+)(\))", bygroups(Keyword, Name.Variable, Keyword)),
+        ],
+        # Handle strings
+        "string": [
+            (
+                r"(\')",
+                String,
+                "#pop",
+            ),
+            include("interp"),
+            (r"[^\']+", String),
+        ],
+        #
+        "assignment": [
+            (r";", Punctuation, "#pop"),
+            include("root"),
+        ],
+        "field_name_quote": [
+            (
+                r"(\")",
+                Keyword,
+                "#pop",
+            ),
+            include("interp"),
+            (r"[^\"]+", Keyword),
+        ],
+        "field_name_bracket": [
+            (
+                r"(\])",
+                Keyword,
+                "#pop",
+            ),
+            include("interp"),
+            (r"[^\]]+", Keyword),
+        ],
+        "function": [(r"\)", Operator.Word, "#pop"), include("root")],
+        "root": [
+            # Whitespace and comments
+            (r"\s+", Text.Whitespace),
+            (r"/\*", Comment.Multiline, "comment"),
+            (r"//.*\n", Comment.Singleline),
+            # variable assignment
+            (
+                r"(let|set)(\s+)",
+                bygroups(
+                    Keyword.Declaration,
+                    Text.Whitespace,
+                ),
+                "assignment",
+            ),
+            # Word operators
+            (
+                words(OPERATORS_LIST["words"], prefix=r"\b", suffix=r"\b"),
+                Operator.Word,
+            ),
+            # Statements
+            (
+                words(
+                    STATEMENT_LIST,
+                    suffix=r"\b",
+                ),
+                Keyword,
+            ),
+            # Table names
+            (r"[a-z]\w*:", Keyword.Declaration),
+            # Constants
+            (words(CONSTANT_LIST, suffix=r"\b"), Keyword.Constant),
+            # Functions
+            (words(SCRIPT_FUNCTIONS, suffix=r"\s*\("), Operator.Word, "function"),
+            # interpolation - e.g. $(variableName)
+            include("interp"),
+            # Quotes denote a field/file name
+            (r'"', Keyword, "field_name_quote"),
+            # Square brackets denote a field/file name
+            (r"\[", Keyword, "field_name_bracket"),
+            # Strings
+            (r"\'", String, "string"),
+            # Numbers
+            include("numerics"),
+            # Operator symbols
+            (words(OPERATORS_LIST["symbols"]), Operator),
+            # Strings denoted by single quotes
+            (r"'.+'", String),
```

Comment:
The `+` quantifier is greedy, so this is going to match everything in `'one string' stuff in between 'another string'`. You need `.+?`. When you fix this, please make sure to cover it in the test.